model/atmosphere/advection/src/icon4py/model/atmosphere/advection/stencils/compute_intermediate_horizontal_tracer_flux_from_cubic_coefficients.py

```diff
@@ -0,0 +1,167 @@
+    z_lsq_coeff_5: fa.CellKField[ta.wpfloat],
+    z_lsq_coeff_6: fa.CellKField[ta.wpfloat],
+    z_lsq_coeff_7: fa.CellKField[ta.wpfloat],
+    z_lsq_coeff_8: fa.CellKField[ta.wpfloat],
+    z_lsq_coeff_9: fa.CellKField[ta.wpfloat],
+    z_lsq_coeff_10: fa.CellKField[ta.wpfloat],
+    z_quad_vector_sum0_1: fa.EdgeKField[ta.vpfloat],
+    z_quad_vector_sum0_2: fa.EdgeKField[ta.vpfloat],
+    z_quad_vector_sum0_3: fa.EdgeKField[ta.vpfloat],
+    z_quad_vector_sum0_4: fa.EdgeKField[ta.vpfloat],
+    z_quad_vector_sum0_5: fa.EdgeKField[ta.vpfloat],
+    z_quad_vector_sum0_6: fa.EdgeKField[ta.vpfloat],
+    z_quad_vector_sum0_7: fa.EdgeKField[ta.vpfloat],
+    z_quad_vector_sum0_8: fa.EdgeKField[ta.vpfloat],
+    z_quad_vector_sum0_9: fa.EdgeKField[ta.vpfloat],
+    z_quad_vector_sum0_10: fa.EdgeKField[ta.vpfloat],
+    patch0_cell_rel_idx_dsl: fa.EdgeKField[gtx.int32],
+) -> fa.EdgeKField[ta.wpfloat]:
+    p_out_e_hybrid_1a = (
+        where(
+            patch0_cell_rel_idx_dsl == 1,
+            z_lsq_coeff_1(E2C[1]),
+            z_lsq_coeff_1(E2C[0]),
+        )
+        * astype(z_quad_vector_sum0_1, wpfloat)
+        + where(
+            patch0_cell_rel_idx_dsl == 1,
+            z_lsq_coeff_2(E2C[1]),
+            z_lsq_coeff_2(E2C[0]),
+        )
+        * astype(z_quad_vector_sum0_2, wpfloat)
+        + where(
+            patch0_cell_rel_idx_dsl == 1,
+            z_lsq_coeff_3(E2C[1]),
+            z_lsq_coeff_3(E2C[0]),
+        )
+        * astype(z_quad_vector_sum0_3, wpfloat)
+        + where(
+            patch0_cell_rel_idx_dsl == 1,
+            z_lsq_coeff_4(E2C[1]),
+            z_lsq_coeff_4(E2C[0]),
+        )
+        * astype(z_quad_vector_sum0_4, wpfloat)
+        + where(
+            patch0_cell_rel_idx_dsl == 1,
+            z_lsq_coeff_5(E2C[1]),
+            z_lsq_coeff_5(E2C[0]),
+        )
+        * astype(z_quad_vector_sum0_5, wpfloat)
+        + where(
+            patch0_cell_rel_idx_dsl == 1,
+            z_lsq_coeff_6(E2C[1]),
+            z_lsq_coeff_6(E2C[0]),
+        )
+        * astype(z_quad_vector_sum0_6, wpfloat)
+        + where(
+            patch0_cell_rel_idx_dsl == 1,
+            z_lsq_coeff_7(E2C[1]),
+            z_lsq_coeff_7(E2C[0]),
+        )
+        * astype(z_quad_vector_sum0_7, wpfloat)
+        + where(
+            patch0_cell_rel_idx_dsl == 1,
+            z_lsq_coeff_8(E2C[1]),
+            z_lsq_coeff_8(E2C[0]),
+        )
+        * astype(z_quad_vector_sum0_8, wpfloat)
+        + where(
+            patch0_cell_rel_idx_dsl == 1,
+            z_lsq_coeff_9(E2C[1]),
+            z_lsq_coeff_9(E2C[0]),
+        )
+        * astype(z_quad_vector_sum0_9, wpfloat)
+        + where(
+            patch0_cell_rel_idx_dsl == 1,
+            z_lsq_coeff_10(E2C[1]),
+            z_lsq_coeff_10(E2C[0]),
+        )
+        * astype(z_quad_vector_sum0_10, wpfloat)
+    )
+
+    return p_out_e_hybrid_1a
+
+
+@gtx.program(grid_type=gtx.GridType.UNSTRUCTURED, backend=backend)
```

Comment:
```suggestion
@gtx.program(grid_type=gtx.GridType.UNSTRUCTURED)
```

Backend argument should go on stencil call, i.e.:
stencil_name.with_backend(backend)(args)